pyfr/readers/native.py

```diff
@@ -166,6 +182,32 @@ def __len__(self):
         return len(self._npf.files)
 
 
+class PyFRH5Reader(PyFRBaseReader):
+    """Does everything that file reader does but with a h5py file"""
+    def __init__(self, fname):
+        self._file = h5py.File(fname, 'r')
+
+    def __getitem__(self, aname):
+        ret = self._file[aname]
+
+        if self._file[aname].shape == ():
+            ret = ret.value
+
+        else:
+            ret = np.array(ret)
+
+        if isinstance(ret, bytes):
+            return ret.decode()
+
+        return ret
```

Comment:
Put this into an `else` block or, if you prefer merge into a single line `return ret.decode() if isinstance(ret, bytes) else ret`.